testing/tests/consensus.rs

```diff
@@ -108,29 +86,13 @@ where
 /// Builds and submits a random proposal for the specified view number
 async fn submit_validating_proposal<
     TYPES: NodeType<ConsensusType = ValidatingConsensus>,
-    I: NodeImplementation<TYPES, Leaf = ValidatingLeaf<TYPES>>,
+    I: TestableNodeImplementation<TYPES, Leaf = ValidatingLeaf<TYPES>>
+        + NodeImplementation<TYPES, Leaf = ValidatingLeaf<TYPES>>,
```

Comment:
Similar to the comment at `fn is_upcoming_validating_leader`, could we remove the `+ NodeImplementation` part? If it's necessary, I'm wondering why we specify the `Leaf` type for `TestableNodeImplementation` here but not in `is_upcoming_validating_leader`.